csi_camera.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include <vector>
+#include <string>
+#include <memory>
+#include <map>
+
+#include <gst/gst.h>
+#include <gst/app/gstappsink.h>
+#include <viam/sdk/components/camera/camera.hpp>
+#include <viam/api/component/camera/v1/camera.grpc.pb.h>
+
+using namespace viam::sdk;
+
+class CSICamera : public Camera {
+private:
+    // Camera
+    bool debug;
+    int width_px = 0;
+    int height_px = 0;
+    int frame_rate = 0;
+    std::string video_path;
+
+    // GST
+    GstElement *pipeline = nullptr;
+    GstBus *bus = nullptr;
+    GstMessage *msg = nullptr;
+    GstSample *sample = nullptr;
+    GstBuffer *buffer = nullptr;
+    GstElement* appsink = nullptr;
+
+public:
+    // Module
+    explicit CSICamera(const std::string name, const AttributeMap attrs);
+    ~CSICamera();
+    void init(const AttributeMap attrs);
+    void init_csi(const std::string pipeline_args);
+    void validate_attrs(const AttributeMap attrs);
+    template <typename T>
+    void set_attr(const AttributeMap& attrs, const std::string& name, T CSICamera::* member, T de);
+
+    // Camera
+    // overrides camera component interface
+    void reconfigure(const Dependencies deps, const ResourceConfig cfg) override;
+    raw_image get_image(const std::string mime_type) override;
+    image_collection get_images() override;
+    AttributeMap do_command(const AttributeMap command) override;
+    point_cloud get_point_cloud(const std::string mime_type) override;
+    std::vector<GeometryConfig> get_geometries() override;
+    properties get_properties() override;
+
+    // GST 
+    // helpers to manage GStreamer pipeline lifecycle
+    std::string create_pipeline() const;
+    void wait_pipeline();
+    void stop_pipeline();
+    void catch_pipeline();
+
+    // Image 
+    // helpers to pull and process images from appsink
+    std::vector<unsigned char> get_csi_image();
+    std::vector<unsigned char> buff_to_vec(GstBuffer *buff);
+    std::vector<unsigned char> get_test_image();
```

Comment:
nit: do these methods need to be public?

Reply:
Not right now. I was leaving these public for unit testing later on.